spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/VectorizedSparkParquetReaders.java

```diff
@@ -51,22 +53,43 @@ public class VectorizedSparkParquetReaders {
 
   private VectorizedSparkParquetReaders() {}
 
-  public static ColumnarBatchReader buildReader(
+  public static VectorizedReader buildReader(
       Schema expectedSchema,
       MessageType fileSchema,
       Map<Integer, ?> idToConstant,
       DeleteFilter<InternalRow> deleteFilter) {
-    return (ColumnarBatchReader)
-        TypeWithSchemaVisitor.visit(
-            expectedSchema.asStruct(),
-            fileSchema,
-            new ReaderBuilder(
-                expectedSchema,
-                fileSchema,
-                NullCheckingForGet.NULL_CHECKING_ENABLED,
-                idToConstant,
-                ColumnarBatchReader::new,
-                deleteFilter));
+    boolean hasCometJar = false;
+    try {
```

Comment:
I would **_consider_** the following mechanism for controlling Comet (just a thought):
- Add an enum called `ReaderType` (or a similar name) with two values: `ICEBERG` and `COMET`.
- Add a new property to `SparkSQLProperties` that would control which reader to use.
  - Something like `spark.sql.iceberg.parquet.reader-type`.
- Parse the SQL conf in `SparkBatch` and pass the reader type to `SparkColumnarReaderFactory`.
- Propagate the reader type to `BaseBatchReader`.
- Add a new method `buildCometReader` to `VectorizedSparkParquetReaders`.
  - I don't think we should use Comet depending on whether the jar is there.
- Chose either `buildReader` or `buildCometReader` based on the reader type in `newParquetIterable`.

Reply:
Thanks a lot for your suggestion! I will change the code accordingly.